Mathlib/CategoryTheory/Sums/Products.lean

```diff
@@ -0,0 +1,148 @@
+    (whiskeringLeft A (A ⊕ A') B).obj (inl_ A A') :=
+ NatIso.ofComponents (fun _ ↦ Iso.refl _)
+
+/-- Composing the forward direction of `functorEquiv` with the second projection is the same as
+ precomposition with `inr_ A A'`. -/
+@[simps!]
+def functorEquivFunctorCompSndIso :
+    (functorEquiv A A' B).functor ⋙ Prod.snd (A ⥤ B) (A' ⥤ B) ≅
+    (whiskeringLeft A' (A ⊕ A') B).obj (inr_ A A') :=
+  NatIso.ofComponents (fun _ ↦ Iso.refl _)
+
+/-- Composing the backward direction of `functorEquiv` with precomposition with `inl_ A A'`.
+ is naturally isomorphic to the first projection. -/
+@[simps!]
+def functorEquivInverseCompWhiskeringLeftInlIso :
+    (functorEquiv A A' B).inverse ⋙ (whiskeringLeft A (A ⊕ A') B).obj (inl_ A A') ≅
+    Prod.fst (A ⥤ B) (A' ⥤ B) :=
+  NatIso.ofComponents (fun _ ↦ Functor.inlCompSum' _ _)
+
+/-- Composing the backward direction of `functorEquiv` with the second projection is the same as
+ precomposition with `inr_ A A'`. -/
+@[simps!]
+def functorEquivInverseCompWhiskeringLeftInrIso :
+    (functorEquiv A A' B).inverse ⋙ (whiskeringLeft A' (A ⊕ A') B).obj (inr_ A A') ≅
+    Prod.snd (A ⥤ B) (A' ⥤ B) :=
+  NatIso.ofComponents (fun _ ↦ Functor.inrCompSum' _ _)
+
+end Sum
+
+variable {A A' B}
+
+namespace NatTrans
+
+/-- A consequence of `functorEquiv`: we can construct a natural transformation of functors
+`A ⊕ A' ⥤ B` from the data of natural transformations of their whiskering with `inl_` and `inr_`. -/
+@[simps!]
+def ofNatTransWhiskerLeftInlInr {F G : A ⊕ A' ⥤ B}
+    (η₁ : Sum.inl_ A A' ⋙ F ⟶ Sum.inl_ A A' ⋙ G) (η₂ : Sum.inr_ A A' ⋙ F ⟶ Sum.inr_ A A' ⋙ G) :
+    F ⟶ G :=
+  (Sum.functorEquiv A A' B).unit.app F ≫
+    (Sum.functorEquiv A A' B).inverse.map ((η₁, η₂) :) ≫
+      (Sum.functorEquiv A A' B).unitInv.app G
+
+@[simp]
+lemma ofNatTransWhiskerLeftInlInr_id {F : A ⊕ A' ⥤ B} :
+    ofNatTransWhiskerLeftInlInr (𝟙 (Sum.inl_ A A' ⋙ F)) (𝟙 (Sum.inr_ A A' ⋙ F)) = 𝟙 F := by
+  aesop_cat
+
+@[simp]
+lemma ofNatTransWhiskerLeftInlInr_comp {F G H : A ⊕ A' ⥤ B}
+    (η₁ : Sum.inl_ A A' ⋙ F ⟶ Sum.inl_ A A' ⋙ G) (η₂ : Sum.inr_ A A' ⋙ F ⟶ Sum.inr_ A A' ⋙ G)
+    (ν₁ : Sum.inl_ A A' ⋙ G ⟶ Sum.inl_ A A' ⋙ H) (ν₂ : Sum.inr_ A A' ⋙ G ⟶ Sum.inr_ A A' ⋙ H) :
+    ofNatTransWhiskerLeftInlInr (η₁ ≫ ν₁) (η₂ ≫ ν₂) = ofNatTransWhiskerLeftInlInr η₁ η₂ ≫
+      ofNatTransWhiskerLeftInlInr ν₁ ν₂ := by
+  aesop_cat
+
+end NatTrans
+
+namespace NatIso
+
+/-- A consequence of `functorEquiv`: we can construct a natural isomorphism of functors
+`A ⊕ A' ⥤ B` from the data of natural isomorphisms of their whiskering with `inl_` and `inr_`. -/
+@[simps]
+def ofNatIsoWhiskerLeftInlInr {F G : A ⊕ A' ⥤ B}
+    (η₁ : Sum.inl_ A A' ⋙ F ≅ Sum.inl_ A A' ⋙ G) (η₂ : Sum.inr_ A A' ⋙ F ≅ Sum.inr_ A A' ⋙ G) :
+    F ≅ G where
+  hom := NatTrans.ofNatTransWhiskerLeftInlInr η₁.hom η₂.hom
+  inv := NatTrans.ofNatTransWhiskerLeftInlInr η₁.inv η₂.inv
+
+lemma ofNatIsoWhiskerLeftInlInr_eq {F G : A ⊕ A' ⥤ B}
```

Comment:
I would suggest putting `ofNatIsoWhiskerLeftInlInr` in the LHS instead of the RHS.